packages/react-components/react-table/src/components/DataGrid/useDataGrid.ts

```diff
@@ -45,11 +50,24 @@ export const useDataGrid_unstable = (props: DataGridProps, ref: React.Ref<HTMLEl
       onSelectionChange,
       selectionMode: selectionMode ?? 'multiselect',
     }),
+    useTableColumnSizing_unstable({
+      onColumnResize,
+      columnSizingOptions,
+      // The selection cell is not part of the columns, therefore its width needs to be subtracted
+      // from the container to make sure the columns don't overflow the table.
+      containerWidthOffset: selectionMode ? -CELL_WIDTH : 0,
+    }),
   ]);
 
   const baseTableState = useTable_unstable(
-    { role: 'grid', as: 'div', noNativeElements: true, ...(navigable && keyboardNavAttr), ...props },
-    ref,
+    {
+      role: 'grid',
+      as: 'div',
+      noNativeElements: true,
+      ...(navigable && keyboardNavAttr),
+      ...props,
+    },
+    useMergedRefs(ref, tableState.tableRef),
```

Comment:
I **think** this is a correct use for this hook, but I'd like a confirmation here.)

Reply:
yes this is correct